templates/vue/src/components/Lightbox/media/MultiContentMedia/PageMenu/MenuGroups.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+  <b-container>
+    <div v-for="(menu, index) in filteredMenuGroups" :key="index" class="py-1">
+      <b-card>
+        <b-card-text
+          v-b-toggle="index === 0 || menu.length === 0 ? `` : `collapse-${index}`"
+          @click="handleMenuTitleClick(index)"
+        >
+          {{ getMenuName(index) }}
+        </b-card-text>
+        <b-collapse :id="`collapse-${index}`" :visible="index === 0" class="mt-2">
+          <ul class="page-menu-item fa-ul">
+            <page-menu-item
+              v-for="row in menu"
+              :key="row.node.id"
+              :node="row.node"
+              :lock-rows="lockRows"
+              :disabled="disabled"
+              @scroll-to="scrollTo"
+              @handle-menu-item-click="handleMenuItemClick"
+            />
+          </ul>
+        </b-collapse>
+      </b-card>
+    </div>
+  </b-container>
+</template>
+
+<script>
+import { mapGetters } from "vuex"
+import PageMenuItem from "./PageMenuItem"
+export default {
+  name: "menu-groups",
+  components: {
+    PageMenuItem,
+  },
+  props: {
+    node: {
+      type: Object,
+      required: true,
+    },
+    rows: {
+      type: Array,
+      required: true,
+    },
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  computed: {
+    ...mapGetters(["getParent"]),
+    lockRows() {
+      return this.node.typeData.lockRows
+    },
+    disabledFrom() {
+      return this.rows.findIndex(row => !row.node.completed)
+    },
+    menuGroups() {
+      const menu = []
+      const mainMenu = []
+      this.rows.forEach(row => {
+        row.node.typeData.isSecondaryNode ? menu.push([row]) : mainMenu.push(row)
+      })
+      menu.unshift(mainMenu)
+      return menu
+    },
+    filteredMenuGroups() {
+      const filteredMenu = []
+      this.menuGroups.forEach((menu, mIndex) => {
+        if (mIndex === 0) {
+          filteredMenu.push(menu)
+        } else {
+          let childrenNodes = menu[0].children
+          const newMenu = []
+          childrenNodes.forEach(childNode => {
+            let newRow = {
+              children: [],
+              node: childNode,
+            }
+            newMenu.push(newRow)
+          })
+          filteredMenu.push(newMenu)
+        }
+      })
+      return filteredMenu
+    },

```

Comment:
If you change this to a single menu group, you can move all of this to the parent.